src/Ansatz/Chain.jl

```diff
@@ -343,6 +343,51 @@ function isrightcanonical(qtn::Chain, site; atol::Real = 1e-12)
     return isapprox(contracted, identity_matrix; atol)
 end
 
+canonize(tn::Chain, args...; kwargs...) = canonize!(deepcopy(tn), args...; kwargs...)
```

Comment:
Do you really need `deepcopy`? This copies all array contents too so it's costly. Doesn't it work with `copy`?

`deepcopy` should be just used in the case that the arrays of a TN are modified in-place. If the TN is modified in-place but the arrays not, then use `copy`.